addons/binding/org.openhab.binding.meterreader/README.md

```diff
@@ -0,0 +1,130 @@
+# MeterReader Binding
+
+The meter reader binding is able to read SML messages (PUSH) and supports IEC 62056-21 modes A,B,C (PULL)and D (PUSH).
+
+To decode SML messages, the binding takes advantage of jSML library by Fraunhofer Institute.
+Please see following page for more informations (German but with good links):
+https://de.wikipedia.org/wiki/Smart_Message_Language
+
+To support IEC 62056-21 protocol the binding uses [j62056 library](https://www.openmuc.org/iec-62056-21/)
+
+
+## Supported Things
+
+This binding supports only one Thing: `meter`
+
+## Discovery
+
+Discovery is not available, as the binding only reads from serial ports.
+
+## Binding Configuration
+
+No binding configuration required.
+
+## Thing Configuration
+
+The MeterReader thing requires the serial port where the meter device is connected and optionally an refresh intervall.
+
+| Parameter | Name | Description | Required | Default |
+|-----------|------|-------------|----------|---------|
+| `port` | The serial port to connect to| URL to use for playing notification sounds, e.g. `/dev/ttyUSB0` | yes | |
+| `refresh` | The refresh interval in seconds | Defines at which interval the values of the meter device shall be read | no | 20 |
+| `mode` | The protocol mode to use | Can be 'SML' (PUSH mode), 'ABC' (PULL)or D (PUSH) | no | SML |
+| `baudrateChangeDelay` | Delay of baudrate change in ms | USB to serial converters often require a delay of up to 250ms after the ACK before changing baudrate (only relevant for 'C' mode) | no | 0 |
+| `baudrate` | (initial) Baudrate | The baudrate of the serial port. If set to 'auto', it will be negotiated with the meter. The default is 300 baud for modes A, B, and C and 2400 baud for mode D, and 9600 baud for SML. | no | AUTO |
+
+The default is 300 baud for modes A, B, and C and 2400 baud for mode D
+
+## Channels
+
+All available OBIS codes which are read out from the device will be created as channels.
+At every read out the channels will be synched with the OBIS codes from the device.
+
+Following conversion from OBIS codes to channel ID is done:
+`.` is replaced by `-` and `:` or `*` is replaced by `#`.
```

Comment:
`#` is reserved for channels of a channel group. If this is the case here, you should also implement a `ChannelGroupTypeProvider`.

Reply:
Changed `#` to `_`